### PR TITLE
AssetMeta - separate connected components

### LINKED ISSUE #279: Export each subcomponent in AssetMeta
It would be nice to have all the subcomponents from AssetMeta available. For instance, I want to only show events (without the information about the asset etc).

### PR DESCRIPTION
**Pull request by xitman78**
Added separate connected components that are used in `AssetMeta`:
- `AssetDetailsPanel`
- `AssetDocumentsPanel`
- `AssetEventsPanel`
- `AssetTimseriesPanel`

Closes #279

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-296.gearboxjs.preview.cogniteapp.com

File: src/hoc/withAssetTimeseries.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      if (prevProps.assetId !== this.props.assetId) {
+        this.loadAssetTimeseries();
+      }
+    }
+
+    async loadAssetTimeseries() {
+      try {
+        const res = await connectPromiseToUnmountState(
+          this,
+          sdk.TimeSeries.list({ assetId: this.props.assetId, limit: 1000 })
```
Comment (mrtraser) [inline src/hoc/withAssetTimeseries.tsx]: The same `limit` prop here

File: src/components/AssetMeta/AssetMeta.tsx [hunk 2]
```diff
@@ -57,220 +41,181 @@ interface AssetMetaProps {
...
-    if (prevProps.assetId !== this.props.assetId) {
-      this.setState({
-        isLoading: true,
-      });
-      this.loadAll(this.props.assetId);
+    if (
+      prevProps.assetId !== this.props.assetId &&
+      !this.includesPanel('details')
+    ) {
+      this.loadAsset(this.props.assetId);
```
[the remark this answers is not in the document]
Comment (xitman78) [reply src/components/AssetMeta/AssetMeta.tsx]: but it's used by other methods in the component

File: src/hoc/withAssetEvents.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      if (prevProps.assetId !== this.props.assetId) {
+        this.loadAssetEvents();
+      }
+    }
+
+    async loadAssetEvents() {
+      try {
+        const res = await connectPromiseToUnmountState(
+          this,
+          sdk.Events.list({ assetId: this.props.assetId, limit: 1000 })
```
[the remark this answers is not in the document]
Comment (mrtraser) [reply src/hoc/withAssetEvents.tsx]: Mentioned in #297

File: src/components/AssetMeta/AssetMeta.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrtraser) [reply src/components/AssetMeta/AssetMeta.tsx]: I meant move `!this.includesPanel('details')` check inside `this.loadAsset`, like:
```typescript
loadAsset() {
    if (this.includesPanel('details')) return;
}
```

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-296.gearboxjs.preview.cogniteapp.com

File: src/components/AssetMeta/AssetMeta.tsx [hunk 4]
```diff
@@ -57,220 +41,181 @@ interface AssetMetaProps {
...
     return (
-      <TabPane
-        tab={this.tabStyle('Details', Object.keys(asset.metadata).length)}
-        key="details"
-      >
-        <DescriptionList
-          valueSet={asset.metadata}
+      <TabPane tab="Details" key="details" forceRender={true}>
+        <AssetDetailsPanel
+          assetId={this.props.assetId}
```
[the remark this answers is not in the document]
Comment (xitman78) [reply src/components/AssetMeta/AssetMeta.tsx]: done

**Pull request merged**